From ac099be0f85e6406f0f092319c4fd4ab8f9b5ce3 Mon Sep 17 00:00:00 2001
From: Rakshit Gondwal <rakshit.gondwal@chainguard.dev>
Date: Wed, 22 Oct 2025 13:15:05 +0530
Subject: [PATCH] fix: use ecdsa p-384 for FIPS compat

This patch removes the usage of ECDSA Ed25519 which is not FIPS supported. It instead uses ECDSA P-384 which supports FIPS.

Signed-off-by: Rakshit Gondwal <rakshit.gondwal@chainguard.dev>
---
 lib/tlsutil/tlsutil.go | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/lib/tlsutil/tlsutil.go b/lib/tlsutil/tlsutil.go
index cbdd05b3b..a7bbbfccd 100644
--- a/lib/tlsutil/tlsutil.go
+++ b/lib/tlsutil/tlsutil.go
@@ -108,9 +108,14 @@ func generateCertificate(commonName string, lifetimeDays int, compatible bool) (
 			pub = pk.Public()
 		}
 	} else {
-		// For sync connections we use Ed25519
-		sigAlgo = x509.PureEd25519
-		pub, priv, err = ed25519.GenerateKey(rand.Reader)
+		// For sync connections we use ECDSA P-384
+		sigAlgo = x509.ECDSAWithSHA384
+		var pk *ecdsa.PrivateKey
+		pk, err = ecdsa.GenerateKey(elliptic.P384(), rand.Reader)
+		if err == nil {
+			priv = pk
+			pub = pk.Public()
+		}
 	}
 	if err != nil {
 		return nil, nil, fmt.Errorf("generate key: %w", err)
-- 
2.39.5 (Apple Git-154)

